src/Microsoft.AspNet.WebUtilities/QueryHelpers.cs

```diff
@@ -20,8 +20,7 @@ public static class QueryHelpers
         /// <returns>The combined result.</returns>
         public static string AddQueryString([NotNull] string uri, [NotNull] string name, [NotNull] string value)
         {
-            bool hasQuery = uri.IndexOf('?') != -1;
-            return uri + (hasQuery ? "&" : "?") + UrlEncoder.Default.UrlEncode(name) + "=" + UrlEncoder.Default.UrlEncode(value);
+            return AddQueryString(uri, new Dictionary<string, string> { { name, value } });
```

Comment:
O I C. That would mean having both the public methods call into a new 3rd private method that worked off of `IEnumerable<KVP<,>>` right? That does seem reasonable to me.